Rename ControlPartType enum class to StyleAppearance

nt1m opened the pull request:
#### 5b6dcd213c1d1503c197db3060322c18b1ee4ae4
<pre>
Rename ControlPartType enum class to StyleAppearance
<a href="https://bugs.webkit.org/show_bug.cgi?id=250067">https://bugs.webkit.org/show_bug.cgi?id=250067</a>
rdar://103865254

Reviewed by Aditya Keerthi.

This makes checks like `style.effectiveAppearance() == ControlPartType::X` less confusing, and emphasizes the connection with the appearance property.
Also, rename ControlPartType::NoControl to StyleAppearance::None.

The ControlPart terminology is preserved for form control drawing code.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isApplePayButton const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::orientation const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
(WebCore::CSSPrimitiveValue::operator StyleAppearance const):
(WebCore::CSSPrimitiveValue::operator ControlPartType const): Deleted.
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleKeydownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::hasVerticalAppearance):
(WebCore::SliderThumbElement::resolveCustomStyle):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::resolveCustomStyle):
(WebCore::SearchFieldCancelButtonElement::resolveCustomStyle):
* Source/WebCore/platform/Theme.cpp:
(WebCore::Theme::baselinePositionAdjustment const):
(WebCore::Theme::controlFont const):
(WebCore::Theme::controlSize const):
(WebCore::Theme::minimumControlSize const):
(WebCore::Theme::controlRequiresPreWhiteSpace const):
(WebCore::Theme::paint):
(WebCore::Theme::inflateControlPaintRect const):
(WebCore::Theme::controlBorder const):
(WebCore::Theme::controlPadding const):
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/ThemeTypes.h:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(WebCore::ThemeAdwaita::controlSize const):
(WebCore::ThemeAdwaita::minimumControlSize const):
(WebCore::ThemeAdwaita::controlBorder const):
(WebCore::ThemeAdwaita::paint):
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/controls/ButtonPart.h:
* Source/WebCore/platform/graphics/controls/ColorWellPart.h:
* Source/WebCore/platform/graphics/controls/ControlPart.cpp:
(WebCore::ControlPart::ControlPart):
* Source/WebCore/platform/graphics/controls/ControlPart.h:
(WebCore::ControlPart::type const):
* Source/WebCore/platform/graphics/controls/ControlPartType.cpp: Removed.
* Source/WebCore/platform/graphics/controls/InnerSpinButtonPart.h:
* Source/WebCore/platform/graphics/controls/MenuListButtonPart.h:
* Source/WebCore/platform/graphics/controls/MenuListPart.h:
* Source/WebCore/platform/graphics/controls/MeterPart.cpp:
(WebCore::MeterPart::MeterPart):
* Source/WebCore/platform/graphics/controls/ProgressBarPart.cpp:
(WebCore::ProgressBarPart::ProgressBarPart):
* Source/WebCore/platform/graphics/controls/SearchFieldCancelButtonPart.h:
* Source/WebCore/platform/graphics/controls/SearchFieldPart.h:
* Source/WebCore/platform/graphics/controls/SliderThumbPart.h:
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::create):
(WebCore::SliderTrackPart::SliderTrackPart):
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.h:
(isType):
* Source/WebCore/platform/graphics/controls/TextAreaPart.h:
* Source/WebCore/platform/graphics/controls/TextFieldPart.h:
* Source/WebCore/platform/graphics/controls/ToggleButtonPart.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawControlPart::type const):
* Source/WebCore/platform/graphics/mac/controls/ButtonMac.mm:
(WebCore::ButtonMac::ButtonMac):
(WebCore::ButtonMac::bezelStyle const):
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.mm:
(WebCore::ControlFactoryMac::createPlatformButton):
(WebCore::ControlFactoryMac::createPlatformToggleButton):
* Source/WebCore/platform/graphics/mac/controls/SliderThumbMac.mm:
(WebCore::SliderThumbMac::SliderThumbMac):
(WebCore::SliderThumbMac::rectForBounds const):
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.mm:
(WebCore::SliderTrackMac::rectForBounds const):
(WebCore::SliderTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/TextAreaMac.mm:
(WebCore::TextAreaMac::TextAreaMac):
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.mm:
(WebCore::ToggleButtonMac::ToggleButtonMac):
(WebCore::ToggleButtonMac::cellSize const):
(WebCore::ToggleButtonMac::cellOutsets const):
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::configureToggleButton):
(WebCore::createToggleButtonCell):
(WebCore::sharedRadioCell):
(WebCore::sharedCheckboxCell):
(WebCore::setUpButtonCell):
(WebCore::button):
(WebCore::ThemeMac::baselinePositionAdjustment const):
(WebCore::ThemeMac::controlFont const):
(WebCore::ThemeMac::controlSize const):
(WebCore::ThemeMac::minimumControlSize const):
(WebCore::ThemeMac::controlBorder const):
(WebCore::ThemeMac::controlPadding const):
(WebCore::ThemeMac::inflateControlPaintRect const):
* Source/WebCore/rendering/RenderAttachment.cpp:
(WebCore::RenderAttachment::shouldDrawBorder const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterStyleChange):
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterLayout):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::createScrollbar):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::menuStyle const):
(RenderMenuList::clientPaddingLeft const):
(RenderMenuList::clientPaddingRight const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):
(WebCore::isAppearanceAllowedForAllElements):
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::createSliderTrackPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
(WebCore::RenderTheme::extractControlStyleForRenderer const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::paintBorderOnly):
(WebCore::RenderTheme::paintDecorations):
(WebCore::RenderTheme::isControlContainer const):
(WebCore::RenderTheme::isControlStyled const):
(WebCore::RenderTheme::supportsFocusRing const):
(WebCore::RenderTheme::isDefault const):
(WebCore::RenderTheme::supportsMeter const):
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::scrollbarControlSizeForPart):
(WebCore::RenderTheme::canPaint const):
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::supportsFocusRing const):
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
(WebCore::RenderThemeAdwaita::paintMenuList):
(WebCore::RenderThemeAdwaita::paintSliderTrack):
(WebCore::RenderThemeAdwaita::adjustSliderThumbSize const):
(WebCore::RenderThemeAdwaita::paintSliderThumb):
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canPaint const):
(WebCore::RenderThemeIOS::adjustedPaintRect const):
(WebCore::RenderThemeIOS::baselinePosition const):
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::canAdjustBorderRadiusForAppearance):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::adjustSliderThumbSize const):
(WebCore::RenderThemeIOS::adjustButtonStyle const):
(WebCore::RenderThemeIOS::supportsBoxShadow const):
(WebCore::RenderThemeIOS::supportsMeter const):
(WebCore::RenderThemeIOS::paintSliderTicks):
(WebCore::RenderThemeIOS::paintSliderTrackWithFormControlRefresh):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canPaint const):
(WebCore::RenderThemeMac::canCreateControlPartForRenderer const):
(WebCore::RenderThemeMac::canCreateControlPartForBorderOnly const):
(WebCore::RenderThemeMac::canCreateControlPartForDecorations const):
(WebCore::RenderThemeMac::isControlStyled const):
(WebCore::RenderThemeMac::adjustRepaintRect):
(WebCore::RenderThemeMac::controlSupportsTints const):
(WebCore::RenderThemeMac::supportsMeter const):
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
(WebCore::RenderThemeMac::adjustSliderThumbSize const):
* Source/WebCore/rendering/RenderThemeWin.cpp:
(WebCore::RenderThemeWin::supportsFocus const):
(WebCore::RenderThemeWin::determineClassicState):
(WebCore::RenderThemeWin::determineState):
(WebCore::RenderThemeWin::getClassicThemeData):
(WebCore::RenderThemeWin::getThemeData):
(WebCore::drawControl):
(WebCore::RenderThemeWin::paintSliderTrack):
(WebCore::RenderThemeWin::adjustSliderThumbSize const):
(WebCore::RenderThemeWin::supportsMeter const):
* Source/WebCore/rendering/RenderThemeWin.h:
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::hasAppearance const):
(WebCore::RenderStyle::hasEffectiveAppearance const):
(WebCore::RenderStyle::appearance const):
(WebCore::RenderStyle::effectiveAppearance const):
(WebCore::RenderStyle::setAppearance):
(WebCore::RenderStyle::setEffectiveAppearance):
(WebCore::RenderStyle::initialAppearance):
* Source/WebCore/style/StyleAppearance.cpp: Added.
(WebCore::operator&lt;&lt;):
* Source/WebCore/style/StyleAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ControlPartType.h.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;ControlPart&gt;::encode):
(IPC::ArgumentCoder&lt;ControlPart&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/258690@main">https://commits.webkit.org/258690@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d2b093a4efbdecb81e2554556a76c90ef8bfd1cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102682 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11805 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35711 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111944 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/172187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106649 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12814 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2711 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94945 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109637 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108458 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9829 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93044 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37488 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24557 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79230 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5262 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25975 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5416 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2424 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11437 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45465 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5969 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7153 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/Headers.cmake:
```diff
@@ -1760,7 +1760,7 @@ set(WebCore_PRIVATE_FRAMEWORK_HEADERS
     platform/graphics/controls/ControlFactory.h
     platform/graphics/controls/ControlPart.h
-    platform/graphics/controls/ControlPartType.h
+    platform/graphics/controls/StyleAppearance.h
```
Needs to be updated with the new path.

pxlcoder commented on Source/WebCore/Sources.txt:
```diff
@@ -2205,7 +2205,7 @@ platform/graphics/WOFFFileFormat.cpp
 platform/graphics/controls/ControlFactory.cpp
 platform/graphics/controls/ControlPart.cpp
-platform/graphics/controls/ControlPartType.cpp
+platform/graphics/controls/StyleAppearance.cpp
```
Needs to be updated with the new path.

pxlcoder reviewed (commented):
> The ControlPart terminology is perserved for form control drawing code.

"preserved"

nt1m committed:
Rename ControlPartType enum class to StyleAppearance
https://bugs.webkit.org/show_bug.cgi?id=250067
rdar://103865254

Reviewed by Aditya Keerthi.

This makes checks like `style.effectiveAppearance() == ControlPartType::X` less confusing, and emphasizes the connection with the appearance property.
Also, rename ControlPartType::NoControl to StyleAppearance::None.

The ControlPart terminology is preserved for form control drawing code.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isApplePayButton const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::orientation const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
(WebCore::CSSPrimitiveValue::operator StyleAppearance const):
(WebCore::CSSPrimitiveValue::operator ControlPartType const): Deleted.
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleKeydownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::hasVerticalAppearance):
(WebCore::SliderThumbElement::resolveCustomStyle):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::resolveCustomStyle):
(WebCore::SearchFieldCancelButtonElement::resolveCustomStyle):
* Source/WebCore/platform/Theme.cpp:
(WebCore::Theme::baselinePositionAdjustment const):
(WebCore::Theme::controlFont const):
(WebCore::Theme::controlSize const):
(WebCore::Theme::minimumControlSize const):
(WebCore::Theme::controlRequiresPreWhiteSpace const):
(WebCore::Theme::paint):
(WebCore::Theme::inflateControlPaintRect const):
(WebCore::Theme::controlBorder const):
(WebCore::Theme::controlPadding const):
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/ThemeTypes.h:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(WebCore::ThemeAdwaita::controlSize const):
(WebCore::ThemeAdwaita::minimumControlSize const):
(WebCore::ThemeAdwaita::controlBorder const):
(WebCore::ThemeAdwaita::paint):
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/controls/ButtonPart.h:
* Source/WebCore/platform/graphics/controls/ColorWellPart.h:
* Source/WebCore/platform/graphics/controls/ControlPart.cpp:
(WebCore::ControlPart::ControlPart):
* Source/WebCore/platform/graphics/controls/ControlPart.h:
(WebCore::ControlPart::type const):
* Source/WebCore/platform/graphics/controls/ControlPartType.cpp: Removed.
* Source/WebCore/platform/graphics/controls/InnerSpinButtonPart.h:
* Source/WebCore/platform/graphics/controls/MenuListButtonPart.h:
* Source/WebCore/platform/graphics/controls/MenuListPart.h:
* Source/WebCore/platform/graphics/controls/MeterPart.cpp:
(WebCore::MeterPart::MeterPart):
* Source/WebCore/platform/graphics/controls/ProgressBarPart.cpp:
(WebCore::ProgressBarPart::ProgressBarPart):
* Source/WebCore/platform/graphics/controls/SearchFieldCancelButtonPart.h:
* Source/WebCore/platform/graphics/controls/SearchFieldPart.h:
* Source/WebCore/platform/graphics/controls/SliderThumbPart.h:
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::create):
(WebCore::SliderTrackPart::SliderTrackPart):
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.h:
(isType):
* Source/WebCore/platform/graphics/controls/TextAreaPart.h:
* Source/WebCore/platform/graphics/controls/TextFieldPart.h:
* Source/WebCore/platform/graphics/controls/ToggleButtonPart.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawControlPart::type const):
* Source/WebCore/platform/graphics/mac/controls/ButtonMac.mm:
(WebCore::ButtonMac::ButtonMac):
(WebCore::ButtonMac::bezelStyle const):
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.mm:
(WebCore::ControlFactoryMac::createPlatformButton):
(WebCore::ControlFactoryMac::createPlatformToggleButton):
* Source/WebCore/platform/graphics/mac/controls/SliderThumbMac.mm:
(WebCore::SliderThumbMac::SliderThumbMac):
(WebCore::SliderThumbMac::rectForBounds const):
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.mm:
(WebCore::SliderTrackMac::rectForBounds const):
(WebCore::SliderTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/TextAreaMac.mm:
(WebCore::TextAreaMac::TextAreaMac):
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.mm:
(WebCore::ToggleButtonMac::ToggleButtonMac):
(WebCore::ToggleButtonMac::cellSize const):
(WebCore::ToggleButtonMac::cellOutsets const):
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::configureToggleButton):
(WebCore::createToggleButtonCell):
(WebCore::sharedRadioCell):
(WebCore::sharedCheckboxCell):
(WebCore::setUpButtonCell):
(WebCore::button):
(WebCore::ThemeMac::baselinePositionAdjustment const):
(WebCore::ThemeMac::controlFont const):
(WebCore::ThemeMac::controlSize const):
(WebCore::ThemeMac::minimumControlSize const):
(WebCore::ThemeMac::controlBorder const):
(WebCore::ThemeMac::controlPadding const):
(WebCore::ThemeMac::inflateControlPaintRect const):
* Source/WebCore/rendering/RenderAttachment.cpp:
(WebCore::RenderAttachment::shouldDrawBorder const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterStyleChange):
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterLayout):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::createScrollbar):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::menuStyle const):
(RenderMenuList::clientPaddingLeft const):
(RenderMenuList::clientPaddingRight const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):
(WebCore::isAppearanceAllowedForAllElements):
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::createSliderTrackPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
(WebCore::RenderTheme::extractControlStyleForRenderer const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::paintBorderOnly):
(WebCore::RenderTheme::paintDecorations):
(WebCore::RenderTheme::isControlContainer const):
(WebCore::RenderTheme::isControlStyled const):
(WebCore::RenderTheme::supportsFocusRing const):
(WebCore::RenderTheme::isDefault const):
(WebCore::RenderTheme::supportsMeter const):
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::scrollbarControlSizeForPart):
(WebCore::RenderTheme::canPaint const):
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::supportsFocusRing const):
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
(WebCore::RenderThemeAdwaita::paintMenuList):
(WebCore::RenderThemeAdwaita::paintSliderTrack):
(WebCore::RenderThemeAdwaita::adjustSliderThumbSize const):
(WebCore::RenderThemeAdwaita::paintSliderThumb):
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canPaint const):
(WebCore::RenderThemeIOS::adjustedPaintRect const):
(WebCore::RenderThemeIOS::baselinePosition const):
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::canAdjustBorderRadiusForAppearance):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::adjustSliderThumbSize const):
(WebCore::RenderThemeIOS::adjustButtonStyle const):
(WebCore::RenderThemeIOS::supportsBoxShadow const):
(WebCore::RenderThemeIOS::supportsMeter const):
(WebCore::RenderThemeIOS::paintSliderTicks):
(WebCore::RenderThemeIOS::paintSliderTrackWithFormControlRefresh):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canPaint const):
(WebCore::RenderThemeMac::canCreateControlPartForRenderer const):
(WebCore::RenderThemeMac::canCreateControlPartForBorderOnly const):
(WebCore::RenderThemeMac::canCreateControlPartForDecorations const):
(WebCore::RenderThemeMac::isControlStyled const):
(WebCore::RenderThemeMac::adjustRepaintRect):
(WebCore::RenderThemeMac::controlSupportsTints const):
(WebCore::RenderThemeMac::supportsMeter const):
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
(WebCore::RenderThemeMac::adjustSliderThumbSize const):
* Source/WebCore/rendering/RenderThemeWin.cpp:
(WebCore::RenderThemeWin::supportsFocus const):
(WebCore::RenderThemeWin::determineClassicState):
(WebCore::RenderThemeWin::determineState):
(WebCore::RenderThemeWin::getClassicThemeData):
(WebCore::RenderThemeWin::getThemeData):
(WebCore::drawControl):
(WebCore::RenderThemeWin::paintSliderTrack):
(WebCore::RenderThemeWin::adjustSliderThumbSize const):
(WebCore::RenderThemeWin::supportsMeter const):
* Source/WebCore/rendering/RenderThemeWin.h:
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::hasAppearance const):
(WebCore::RenderStyle::hasEffectiveAppearance const):
(WebCore::RenderStyle::appearance const):
(WebCore::RenderStyle::effectiveAppearance const):
(WebCore::RenderStyle::setAppearance):
(WebCore::RenderStyle::setEffectiveAppearance):
(WebCore::RenderStyle::initialAppearance):
* Source/WebCore/style/StyleAppearance.cpp: Added.
(WebCore::operator<<):
* Source/WebCore/style/StyleAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ControlPartType.h.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<ControlPart>::encode):
(IPC::ArgumentCoder<ControlPart>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/258690@main
